src/main/java/ch/kalunight/zoe/service/InfoPanelRefresher.java

```diff
@@ -1,165 +1,430 @@
+
+    for(DTO.GameInfoCard gameInfoCard : gameInfoCards) {
+      switch(gameInfoCard.gamecard_status) {
+      case IN_CREATION:
+        GameInfoCardRepository.updateGameInfoCardStatusWithId(gameInfoCard.gamecard_id, GameInfoCardStatus.IN_TREATMENT);
+
+        List<DTO.LeagueAccount> accountsLinked = LeagueAccountRepository
+            .getLeaguesAccountsWithGameCardsId(gameInfoCard.gamecard_id);
+
+        DTO.LeagueAccount account = accountsLinked.get(0);
+
+        DTO.CurrentGameInfo currentGame = CurrentGameInfoRepository.getCurrentGameWithLeagueAccountID(account.leagueAccount_id);
+
+        ServerData.getInfocardsGenerator().execute(
+            new InfoCardsWorker(server, infochannel, accountsLinked.get(0), currentGame, gameInfoCard));
+        break;
+      case IN_WAIT_OF_DELETING:
+        GameInfoCardRepository.deleteGameInfoCardsWithId(gameInfoCard.gamecard_id);
+        deleteDiscordInfoCard(server.serv_guildId, gameInfoCard);
+        break;
+      default:
+        break;
+      }
+    }
+  }
+
+  private void refreshInfoPanel(DTO.InfoChannel infoChannelDTO) throws SQLException {
+    ArrayList<String> infoPanels = CommandEvent.splitMessage(refreshPannel());
+
+    List<DTO.InfoPanelMessage> infoPanelMessages = InfoChannelRepository.getInfoPanelMessages(server.serv_guildId);
+
+    if(infoPanels.size() < infoPanelMessages.size()) {
+      int nbrMessageToDelete = infoPanelMessages.size() - infoPanels.size();
+      for(int i = 0; i < nbrMessageToDelete; i++) {
+        DTO.InfoPanelMessage infoPanelMessage = infoPanelMessages.get(i);
+        Message message = infochannel.retrieveMessageById(infoPanelMessage.infopanel_messageId).complete();
+        message.delete().queue();
+        InfoChannelRepository.deleteInfoPanelMessage(infoPanelMessage.infopanel_id);
+      }
+
+    } else {
+      int nbrMessageToAdd = infoPanels.size() - infoPanelMessages.size();
+      for(int i = 0; i < nbrMessageToAdd; i++) {
+        Message message = infochannel.sendMessage(LanguageManager.getText(server.serv_language, "loading")).complete();
+        InfoChannelRepository.createInfoPanelMessage(infoChannelDTO.infoChannel_id, message.getIdLong());
+      }
+    }
+
+    infoPanelMessages.clear();
+    infoPanelMessages.addAll(InfoChannelRepository.getInfoPanelMessages(server.serv_guildId));
+
+    for(int i = 0; i < infoPanels.size(); i++) {
+      DTO.InfoPanelMessage infoPanel = infoPanelMessages.get(i);
+        infochannel.retrieveMessageById(infoPanel.infopanel_messageId).complete().editMessage(infoPanels.get(i)).queue();
+    }
+  }
+
+  private List<DTO.GameInfoCard> refreshGameCardStatus() throws SQLException {
+    List<DTO.GameInfoCard> gameInfoCards = GameInfoCardRepository.getGameInfoCards(server.serv_guildId);
+
+    for(DTO.GameInfoCard gameInfoCard : gameInfoCards) {
+      if(gameInfoCard.gamecard_status == GameInfoCardStatus.IN_WAIT_OF_MATCH_ENDING) {
+        if(gameInfoCard.gamecard_fk_currentgame == 0) {

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/manual/KaluNight/Zoe-Discord-Bot/pullRequest?prid=4739924)